backend/core/src/listings/listings.service.ts

```diff
@@ -36,12 +40,61 @@ export class ListingsService {
       query.andWhere("property.neighborhood = :neighborhood", { neighborhood: params.neighborhood })
     }
 
-    let listings = await query.getMany()
+    let currentPage: number = params.page
+    let itemsPerPage: number = params.limit
 
+    let itemCount: number, totalItemsCount: number, totalPages: number
+    let listings: Listing[]
+
+    if (currentPage > 0 && itemsPerPage > 0) {
+      // Calculate the number of listings to skip (because they belong to lower page numbers)
+      const skip = (currentPage - 1) * itemsPerPage
+      query = query.skip(skip).take(itemsPerPage)
+
+      listings = await query.getMany()
+
+      itemCount = listings.length
+
+      // Issue a separate "COUNT(*) from listings;" query to get the total listings count.
+      totalItemsCount = await this.repository.count()
+      totalPages = Math.floor(totalItemsCount / itemsPerPage)
+    } else {
+      // If currentPage or itemsPerPage aren't specified (or are invalid), issue the SQL query to
+      // get all listings (no pagination).
+      listings = await query.getMany()
+
+      currentPage = 1
+      totalPages = 1
+      itemCount = listings.length
+      itemsPerPage = listings.length
+      totalItemsCount = listings.length
+    }
+
+    // Sort units and preferences.
+    // This step was removed from the SQL query because it interferes with pagination
+    // (See https://github.com/CityOfDetroit/affordable-housing-app/issues/88#issuecomment-865329223)
+    listings.forEach((listing) => {
+      listing.property.units.sort((a, b) => a.maxOccupancy - b.maxOccupancy)
+      listing.preferences.sort((a, b) => a.ordinal - b.ordinal)
+    })
+
+    // TODO: remove jsonpath filtering (it doesn't play well with pagination)
```

Comment:
Can you explain more about this?

Reply:
Yeah sorry, the comment was too short - I updated it to say that if the jsonpath filtering removes any listings, the pagination metadata that's returned will be incorrect.

Reply:
Just so I am 100% understanding this - is this jsonpath filtering something that Exygy has built in Bloom? Could we later incorporate that into the DB query so it still works instead of post-DB query? Or does the pagination library fail if you try to limit the query?

Reply:
We chatted about this offline, recording a couple notes:
- Exygy built in the jsonpath filtering support, but they don't seem to be using it anywhere outside of tests
- We don't see a use for `jsonpath` since our SQL-level filtering will give us all the filtering capabilities we want
- This change leaves `jsonpath` filtering in place, though, so that we can think more about if we want to remove it later, or maybe keep it for the sake of keeping this part of our codebase similar to Exygy's

Reply:
I filed https://github.com/CityOfDetroit/bloom/issues/135 and pointed the TODO at that